pkg/testutil/prometheus.go

```diff
@@ -59,11 +62,25 @@ func NewPrometheus(address string) (*Prometheus, error) {
 
 	return &Prometheus{
 		dir:  db.Dir(),
-		addr: address,
 		db:   db,
+		addr: "<prometheus-not-started>",
 	}, nil
 }
 
+func getFreePort() (int, error) {
+	addr, err := net.ResolveTCPAddr("tcp", ":0")
+	if err != nil {
+		return 0, err
+	}
+
+	l, err := net.ListenTCP("tcp", addr)
+	if err != nil {
+		return 0, err
+	}
+	defer l.Close()
+	return l.Addr().(*net.TCPAddr).Port, nil
```

Comment:
Mh, we still have a chance to lose the port again in between. But there's no real alternative I suppose. Should work well enough anyway.

We could pass Prometheus `:0` but I believe it doesn't expose the port it ended up with anywhere.

Reply:
Exactly that is the problem..